src/pymmcore_widgets/_group_preset_widget/_add_first_preset_widget.py

```diff
@@ -0,0 +1,171 @@
+        wdg_layout.addWidget(bottom_wdg)
+
+        main_layout.addWidget(wdg)
+
+    def _create_top_wdg(self) -> QGroupBox:
+        wdg = QGroupBox()
+        wdg_layout = QHBoxLayout()
+        wdg_layout.setSpacing(5)
+        wdg_layout.setContentsMargins(5, 5, 5, 5)
+        wdg.setLayout(wdg_layout)
+
+        lbl_sizepolicy = QSizePolicy(QSizePolicy.Fixed, QSizePolicy.Fixed)
+
+        gp_lbl = QLabel(text="Group:")
+        gp_lbl.setSizePolicy(lbl_sizepolicy)
+        group_name_lbl = QLabel(text=f"{self._group}")
+        group_name_lbl.setSizePolicy(lbl_sizepolicy)
+
+        ps_lbl = QLabel(text="Preset:")
+        ps_lbl.setSizePolicy(lbl_sizepolicy)
+        self.preset_name_lineedit = QLineEdit()
+        self.preset_name_lineedit.setText(f"{self._preset}")
+
+        spacer = QSpacerItem(30, 10, QSizePolicy.Fixed, QSizePolicy.Fixed)
+
+        wdg_layout.addWidget(gp_lbl)
+        wdg_layout.addWidget(group_name_lbl)
+        wdg_layout.addItem(spacer)
+        wdg_layout.addWidget(ps_lbl)
+        wdg_layout.addWidget(self.preset_name_lineedit)
+
+        return wdg
+
+    def _create_bottom_wdg(self) -> QWidget:
+
+        wdg = QWidget()
+        wdg_layout = QHBoxLayout()
+        wdg_layout.setSpacing(5)
+        wdg_layout.setContentsMargins(0, 0, 0, 0)
+        wdg.setLayout(wdg_layout)
+
+        self.apply_button = QPushButton(text="Create Preset")
+        self.apply_button.setSizePolicy(
+            QSizePolicy(QSizePolicy.Fixed, QSizePolicy.Fixed)
+        )
+        self.apply_button.clicked.connect(self._create_first_preset)
+
+        spacer = QSpacerItem(10, 10, QSizePolicy.Expanding, QSizePolicy.Fixed)
+
+        wdg_layout.addItem(spacer)
+        wdg_layout.addWidget(self.apply_button)
+
+        return wdg
+
+    def _populate_table(self) -> None:
+
+        self.table.clearContents()
+
+        self.table.setRowCount(len(self._dev_prop_val_list))
+        for idx, (dev, prop, _) in enumerate(self._dev_prop_val_list):
+            item = QTableWidgetItem(f"{dev}-{prop}")
+            wdg = PropertyWidget(dev, prop, core=self._mmc)
+            wdg._value_widget.valueChanged.disconnect()  # type: ignore
+            self.table.setItem(idx, 0, item)
+            self.table.setCellWidget(idx, 1, wdg)
+
+    def _create_first_preset(self) -> None:
+
+        dev_prop_val = []
+        for row in range(self.table.rowCount()):
+            device_property = self.table.item(row, 0).text()
+            dev = device_property.split("-")[0]
+            prop = device_property.split("-")[1]
+            value = str(self.table.cellWidget(row, 1).value())
+            dev_prop_val.append((dev, prop, value))
+
+        self._preset = self.preset_name_lineedit.text()
+
+        with block_core(self._mmc.events):
+            for d, p, v in dev_prop_val:
+                self._mmc.defineConfig(self._group, self._preset, d, p, v)
+
+        self._mmc.events.configDefined.emit(self._group, self._preset, d, p, v)
```

Comment:
when playing around with the example a bit:
1. I clicked the plus button
2. I gave a name to the group and clicked "create new group" (without adding any properties
3. I clicked preset ... and got an error:


```pytb
Traceback (most recent call last):
  File "/Users/talley/dev/self/pymmcore-widgets/src/pymmcore_widgets/_group_preset_widget/_add_first_preset_widget.py", line 151, in _create_first_preset
    self._mmc.events.configDefined.emit(self._group, self._preset, d, p, v)
UnboundLocalError: local variable 'd' referenced before assignment
```

so the user needs to somehow be prevented from getting to that point